Authentication/SignUp/View/SignupView.swift

```diff
@@ -33,96 +34,120 @@ public extension SignupViewType {
 /** Default signup view. */
 internal final class SignupView: UIView, SignupViewType, NibLoadable {
     
+    // - Warning: This delegate must be set before calling the `render` function.
     internal var delegate: SignupViewDelegate! //swiftlint:disable:this weak_delegate
     
+    internal var state: SignupViewState = (email: (selected: false, content: .initialEmpty),
+                                           password: (selected: false, content: .initialEmpty),
+                                           username: (selected: false, content: .initialEmpty),
+                                           confirmPassword: (selected: false, content: .initialEmpty),
+                                           signUpButton: (executing: false, enabled: false))
+    
+    internal let tapRecognizer = UITapGestureRecognizer()
+    
+    override func awakeFromNib() {
+        addGestureRecognizer(tapRecognizer)
+        tapRecognizer.reactive.stateChanged.observeValues { [unowned self] _ in
+            self.endEditing(true)
+        }
+    }
+    
     internal var titleLabel: UILabel { return titleLabelOutlet }
     @IBOutlet weak var titleLabelOutlet: UILabel! { didSet { titleLabel.text = titleText } }
     
+// MARK: - Textfields and errors
     internal var usernameTextField: UITextField? { return usernameTextFieldOutlet }
     @IBOutlet weak var usernameTextFieldOutlet: UITextField! { didSet { usernameTextFieldOutlet.placeholder = usernamePlaceholderText } }
+    internal var usernameValidationMessageLabel: UILabel? { return usernameValidationMessageLabelOutlet }
```

Comment:
Because we can't conform to a protocol that expects a `UILabel?` with a `UILabel!` or `UILabel`. The same happens with protocol `UILabel` when having `UILabel!`. Sadly.